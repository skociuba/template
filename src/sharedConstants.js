import {routes} from './routesConstants';

const shared = {
  routes: routes,
  header: {
    menu: {
      primary: [
        {name: 'Main', to: '/main-page'},
        {name: 'FrontendPaginationSorting', to: '/frontend-pagination-sorting'},
        {name: 'StaticTable', to: '/static-table-page'},
        {name: 'BackendControl', to: '/backend-control-page'},
        {name: 'FrontendControl', to: '/frontend-control-page'},
      ],
    },
  },
  mapping: [{value: 'testData', label: 'MappedData'}],
  statuses: {
    B: {
      title: 'BUY',
    },
    S: {
      title: 'SELL',
    },
  },
  names: {
    '603be08a2987d5267f590d2f': {
      title: '603be08a2987d5267f590d2f',
      isSelected: false,
    },
    '603be21b2987d50c34590d38': {
      title: '603be21b2987d50c34590d38',
      isSelected: false,
    },
    all: {
      title: 'all',
      isSelected: false,
    },
  },
};

export {shared};
