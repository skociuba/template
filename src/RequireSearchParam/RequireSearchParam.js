import React from 'react';
import {Navigate, useSearchParams} from 'react-router-dom';
import {shared} from 'sharedConstants';

const paths = {...shared.routes};

const RequireSearchParam = ({children, path}) => {
  const [searchParam] = useSearchParams();

  if (path === paths.testPage.root) {
    try {
      const market = searchParam.get('market');
      const productAlternativeClassificationCode = searchParam.get(
        'productAlternativeClassificationCode',
      );
      const productAlternativeNumber = searchParam.get('productAlternativeNumber');
      const productTypeCode = searchParam.get('productTypeCode');

      if (
        productTypeCode &&
        market &&
        productAlternativeClassificationCode &&
        productAlternativeNumber
      ) {
        sessionStorage.setItem(
          productAlternativeNumber,
          JSON.stringify({
            productTypeCode,
            market,
            productAlternativeClassificationCode,
            productAlternativeNumber,
          }),
        );
        return <Navigate to={`${paths.testPage.root}/${productAlternativeNumber}`} />;
      } else {
        return <Navigate to={paths.mainPage.root} />;
      }
    } catch (error) {
      console.error('session storage is unavailable', error);
      return <Navigate to={paths.mainPage.root} />;
    }
  }

  return children;
};

export default RequireSearchParam;
