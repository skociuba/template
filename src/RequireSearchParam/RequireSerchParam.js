import React from 'react';
import {Navigate, useSearchParams} from 'react-router-dom';
import {shared} from 'sharedConstants';

const paths = {...shared.routes};

const RequireSearchParam = ({children, path}) => {
  const [searchParam] = useSearchParams();

  if (path === paths.testPage.root) {
    try {
      const paramForUrl = searchParam.get('paramForUrl');
      const paramForComponent = searchParam.get('paramForComponent');

      if (paramForComponent && paramForUrl) {
        sessionStorage.setItem(
          paramForUrl,
          JSON.stringify({
            paramForComponent,
            paramForUrl,
          }),
        );
        return <Navigate to={`${paths.testPage.root}/${paramForUrl}`} />;
      } else {
        return <Navigate to={paths.mainPage.root} />;
      }
    } catch (error) {
      console.error('session storage is unavailable', error);
      return <Navigate to={paths.mainPage.root} />;
    }
  }

  return children;
};

export default RequireSearchParam;
