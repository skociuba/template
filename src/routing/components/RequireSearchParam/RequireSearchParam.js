import React from 'react';
import {Navigate, useSearchParams} from 'react-router-dom';
import {shared} from 'sharedConstants';
import PropTypes from 'prop-types';

const paths = {...shared.routes};

const RequireSearchParam = ({children, path}) => {
  const [searchParam] = useSearchParams();

  if (path === paths.testPage.root) {
    try {
      const paramForUrl = searchParam.get('paramForUrl');
      const paramForComponent = searchParam.get('paramForComponent');
      if (paramForComponent && paramForUrl) {
        sessionStorage.setItem(
          paramForUrl,
          JSON.stringify({
            paramForUrl,
            paramForComponent,
          }),
        );
        return <Navigate to={`${paths.testPage.root}/${paramForUrl}`} />;
      } else {
        return <Navigate to={paths.mainPage.root} />;
      }
    } catch (error) {
      console.error('session storage is unavailable', error);
      return <Navigate to={paths.mainPage.root} />;
    }
  }

  return children;
};

RequireSearchParam.propTypes = {
  children: PropTypes.element.isRequired,
  path: PropTypes.string.isRequired,
};

export default RequireSearchParam;
