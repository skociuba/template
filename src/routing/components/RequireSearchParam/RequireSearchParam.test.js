import renderWithRouter from 'utils/tests/renderWithRouter';
import {shared} from 'sharedConstants';

import RequireSearchParam from './RequireSearchParam';
const paths = {...shared.routes};

const defaultProps = {
  children: <div>Page</div>,
  path: paths.mainPage.root,
};

describe('<RequireSearchParam/>', () => {
  it('render without crashing', () => {
    const {container} = renderWithRouter(<RequireSearchParam {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
