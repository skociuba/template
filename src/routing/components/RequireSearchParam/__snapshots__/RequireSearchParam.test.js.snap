// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequireSearchParam/> render without crashing 1`] = `
<div>
  Page
</div>
`;
